framework/src/main/java/io/github/kgress/scaffold/AutomationWait.java

```diff
@@ -234,11 +237,21 @@ public Boolean waitForElementToNotHaveClass(BaseWebElement element, String class
      */
     public Boolean waitUntilPageIsLoaded(Long setTempTimeout) {
         var domReadyStateScript = "return document.readyState";
-        return waitForCustomCondition(page -> getWebDriverWrapper()
-                .getJavascriptExecutor()
-                .executeScript(domReadyStateScript)
-                .equals("complete"),
-                setTempTimeout);
+
+        try {
+            return waitForCustomCondition(page -> getWebDriverWrapper()
+                            .getJavascriptExecutor()
+                            .executeScript(domReadyStateScript)
+                            .equals("complete"),
+                    setTempTimeout);
+        } catch (WebDriverException e) {
+            if (e.getMessage().contains("target frame detached")) {
+                AutomationUtils.sleep(200);
+                return waitUntilPageIsLoaded(setTempTimeout);
+            } else {
+                throw(e);
+            }
+        }
```

Comment:
I think there should be a safe guard here.  At the very least, I think having a number of retries up to a max then failing would be a good idea.

Reply:
I think a test or a few  to verify behavior is a good idea too.

Reply:
I've added a retry limit and tests for that limit